waku/v2/protocol/waku_message.nim

```diff
@@ -13,6 +13,9 @@ import
 when defined(rln):
   import waku_rln_relay/waku_rln_relay_types
 
+const
+  MaxWakuMessageSize* = 1024 * 1024 # In bytes. Corresponds to PubSub default
```

Comment:
Does libp2p have a hardcoded const/variable corresponding to this default? thought might be good to use that one to make sure our approximation is always consistent with the most recent default size of pubsub messages.

Reply:
> I think the solution is good enough given that the size of a waku message always meets the pubsub message size limit. But if this assumption does not hold then there would be the probability of running to the same error (hit the max allowed stream size).

Right. I agree, so I've ensured that we initialise PubSub with this max value (overriding the default max message size). This way, even if for some unlikely reason the underlying PubSub default changes it won't affect any of the Waku protocols. It's a bit hidden in between the other changes, but here I initialise PubSub with our `MaxWakuMessageSize`: https://github.com/status-im/nim-waku/blob/d4eddfced48c1b6e323d495948eb6b8d5c35bd52/waku/v2/node/wakunode2.nim#L644

> Does libp2p have a hardcoded const/variable corresponding to this default? thought might be good to use that one to make sure our approximation is always consistent with the most recent default size of pubsub messages.

It does, but since not all our protocols are PubSub protocols I thought it better to define our own maximum, informed by the PubSub default, which we then apply to all protocols including PubSub (see initialisation line above). This way we're in control of any changes.